### PR TITLE
Mouse Genome Support

### PR DESCRIPTION
**Pull request by apeltzer**
This is starting to add support for Mouse Genomes data to Sarek. 

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/sarek branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/sarek)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

I will work on this ASAP, test locally and then provide both testing data AND the required reference data for iGenomes wherever possible / necessary.

### CODE REVIEW

File: conf/igenomes.config [hunk 1]
```diff
@@ -47,5 +47,24 @@ params {
       snpeffDb               = "GRCh38.86"
       vepCacheVersion        = "95"
     }
+    'GRCm38' {
+      acLoci                 = "${params.igenomes_base}/Mus_musculus/Annotation/ASCAT/fake.loci"
```
Comment (MaxUlysse) [inline conf/igenomes.config]: I think we can remove fake files in configs, and directly not allow people to use ASCAT if no files are available.

File: conf/igenomes.config [hunk 2]
```diff
@@ -47,5 +47,24 @@ params {
...
+      acLociGC               = "${params.igenomes_base}/Mus_musculus/Annotation/ASCAT/fake.loci.gc"
+      bwaIndex               = "${params.igenomes_base}/Mus_musculus/Sequence/BWAIndex/genome.fa.{amb,ann,bwt,pac,sa}"
```
Comment (MaxUlysse) [inline conf/igenomes.config]: I notice that we don't have the same approach in the chipseq pipeline, maybe that's something that we want to harmonize at the nf-core level?
we should discuss about such matters on slack

File: conf/igenomes.config [hunk 3]
```diff
@@ -47,5 +47,24 @@ params {
...
+      chrDir                 = "${params.igenomes_base}/Mus_musculus/Sequence/Chromosomes"
+      chrLength              = "${params.igenomes_base}/Mus_musculus/Sequence/Length/Homo_sapiens_assembly38.len"
+      dbsnp                  = "${params.igenomes_base}/Mus_musculus/Annotation/mgp.v6.merged.norm.snp.indels.sfiltered.vcf.gz"
+      dbsnpIndex             = "${params.igenomes_base}/Mus_musculus/Annotation/mgp.v6.merged.norm.snp.indels.sfiltered.vcf.gz.tbi"
+      dict                   = "${params.igenomes_base}/Mus_musculus/Sequence/WholeGenomeFasta/genome.dict"
+      fasta                  = "${params.igenomes_base}/Mus_musculus/Sequence/WholeGenomeFasta/genome.fa"
+      fastaFai               = "${params.igenomes_base}/Mus_musculus/Sequence/WholeGenomeFasta/genome.fai"
+      germlineResource       = "${params.igenomes_base}/Mus_musculus/Annotation/GermlineResource/gnomAD.r2.1.1.GRCh38.PASS.AC.AF.only.vcf.gz"
```
Comment (MaxUlysse) [inline conf/igenomes.config]: We have more info on that file here: https://github.com/nf-core/sarek/blob/dev/docs/usage.md#--germlineresource

File: conf/genomes.config [hunk 4]
```diff
@@ -27,6 +27,7 @@ params {
       knownIndelsIndex      = "${params.genomes_base}/{1000G_phase1,Mills_and_1000G_gold_standard}.indels.b37.vcf.idx"
       snpeffDb              = "GRCh37.75"
       vepCacheVersion       = "95"
+      species               = 'homo_sapiens'
```
Comment (MaxUlysse) [inline conf/genomes.config]: why single quotes?
Or why do we have double quotes for the rest?

Comment (apeltzer) [reply conf/genomes.config]: Ok, I adjusted them in both `igenomes.config` and `genomes.config` to keep things consistent :-)

File: docs/reference.md [hunk 5]
```diff
@@ -5,7 +5,7 @@
 Sarek is using [AWS iGenomes](https://ewels.github.io/AWS-iGenomes/), which facilitate storing and sharing references.
 Sarek currently uses `GRCh38` by default.
 Both `GRCh37` and `GRCh38` are available with `--genome GRCh37` or `--genome GRCh38` respectively with any profile using the `conf/igenomes.config` file, or you can specify it with `-c conf/igenomes.config`.
```
Comment (MaxUlysse) [inline docs/reference.md]:
```suggestion
`GRCh37`,  `GRCh38` and `GRCm38 `are available with `--genome GRCh37`, `--genome GRCh38` or `--genome GRCm38` respectively with any profile using the `conf/igenomes.config` file, or you can specify it with `-c conf/igenomes.config`.
```

File: docs/reference.md [hunk 6]
```diff
@@ -5,7 +5,7 @@
...
-Use `--genome smallGRCh37` to map against a small reference genome based on GRCh37.
+Use `--genome smallGRCh37` to map against a small reference genome based on GRCh37. Furthermore, mouse genome data can be analyzed using `--genome GRCm38`.
```
Comment (MaxUlysse) [inline docs/reference.md]:
```suggestion
Use `--genome smallGRCh37` to map against a small reference genome based on GRCh37.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -21,6 +21,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - [#41](https://github.com/nf-core/sarek/pull/41) - Update `qualimap` from `2.2.2b` to `2.2.2c`
 - [#41](https://github.com/nf-core/sarek/pull/41), [#55](https://github.com/nf-core/sarek/pull/55) - Update `tiddit` from `2.7.1` to `2.8.1`
 - [#41](https://github.com/nf-core/sarek/pull/41) - Update `vcfanno` from `0.3.1` to `0.3.2`
+- [#46](https://github.com/nf-core/sarek/pull/46) - Add location to abstracts.
```
Comment (MaxUlysse) [inline CHANGELOG.md]: I now realize the issue, #46 and #52 should be in _Added_, not in _Changed_

Comment (apeltzer) [reply CHANGELOG.md]: Fixed that :-)

File: CHANGELOG.md [hunk 8]
```diff
@@ -21,6 +22,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
```
Comment (MaxUlysse) [inline CHANGELOG.md]:
```suggestion
```

File: conf/igenomes.config [hunk 9]
```diff
@@ -44,8 +45,25 @@ params {
...
-      vepCacheVersion        = "95"
+      snpeffDb               = 'GRCh38.86'
+      vepCacheVersion        = '95'
+      species                = 'homo_sapiens'
+    }
+    'GRCm38' {
+      bwaIndex               = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa.{amb,ann,bwt,pac,sa}"
+      chrDir                 = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Chromosomes"
+      chrLength              = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Length/GRCm38.len"
+      dbsnp                  = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/MouseGenomeProject/Annotation/mgp.v5.merged.snps_all.dbSNP142.vcf.gz"
```
Comment (drpatelh) [inline conf/igenomes.config]: Where did you download these files from  @apeltzer ? Looks like they are the latest but may be worth documenting that somewhere e.g.

`dbSNP` files for `GRCm38` were downloaded from `ftp://ftp-mouse.sanger.ac.uk/REL-1505-SNPs_Indels/` on [date].

Comment (drpatelh) [reply conf/igenomes.config]: These files are quite difficult to find so worth being a bit more transparent whilst things are fresh!

Comment (apeltzer) [reply conf/igenomes.config]: I've added a PR to AWS-iGenomes where I'm adding these already:

https://github.com/ewels/AWS-iGenomes/pull/7

Didn't know exactly where this info should be dropped, but that felt like a logical thing to do as others not using Sarek might use the files too and can find the information about where I got these files from there then, too ? 

The `.fai` files and `.dict` match - double checked that and also ran local test data (~50GB) with these change to check that everything is fine 👍

Comment (MaxUlysse) [reply conf/igenomes.config]: That should all be here: https://github.com/ewels/AWS-iGenomes/pull/7

**Pull request merged**